app/xlist/scripts/HomeController.js

```diff
@@ -86,13 +86,15 @@ angular
 
     var THRESHOLD = 0.25;
 
-    var findNear = function(location) {
+    var findNearAndPassLocation = function(location) {
       _.each($scope.pairs, function(pair) {
         var distance = getDistance(location, pair.geoList.location);
         if (distance < THRESHOLD) {
           pushNear(pair);
         }
       });
+      // used to pass location to location controller
+      supersonic.data.channel('location').publish(location);
```

Comment:
I don't think this gives the behavior you're looking for. This seems like it will continue to send location updates to the LocationController, causing the map initialization code to trigger each time. I think it makes more sense to either pass the location as a supersonic view parameter or refactor getLocation into a service and call it in LocationController.